web/src/navbar_alerts.ts

```diff
@@ -251,6 +272,26 @@ export function initialize(): void {
         },
     );
 
+    $(".time-zone-update").on("click", function (e) {
+        e.preventDefault();
+        $(this).closest(".alert").hide();
+        void channel.patch({
+            url: "/json/settings",
+            data: {timezone: browser_time_zone},
+        });
+        $(window).trigger("resize");
```

Comment:
Why does this trigger a resize? It also seems a bit odd that we don't check if the API request succeeded; we probably do want to support an error of some sort. (Probably including only hiding the alert if the request succeeded, in a `success` handler).

Reply:
It appears that in the existing code, any change to the navbar alert is followed by a resize. If a resize is not triggered, it leaves the page in a glitchy state.

I have changed it to only hide the alert if the request succeeds.

Reply:
And for the error, maybe we can show a toast of some sort? In fact, we can show a toast for both success and error.

Reply:
I have started a [discussion](https://chat.zulip.org/#narrow/channel/101-design/topic/Showing.20a.20feedback.20widget.20after.20a.20prompt.20to.20update.20timezone) regarding how to handle the errors.